idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -0,0 +1,1995 @@
+                    )
+                    calculate_variable_from_constraint(
+                        o.costing.bare_erected_cost, o.costing.bare_erected_cost_eq
+                    )
+                    calculate_variable_from_constraint(
+                        o.costing.total_plant_cost, o.costing.total_plant_cost_eq
+                    )
+
+                elif o.costing.library in ["PP", "ASU"]:
+                    for key in o.costing.bare_erected_cost.keys():
+                        calculate_variable_from_constraint(
+                            o.costing.bare_erected_cost[key],
+                            o.costing.bare_erected_cost_eq[key],
+                        )
+                        calculate_variable_from_constraint(
+                            o.costing.total_plant_cost[key],
+                            o.costing.total_plant_cost_eq[key],
+                        )
+
+    def display_total_plant_costs(fs):
+        print("-----Total Plant Costs-----")
+        for o in fs.component_objects(descend_into=False):
+            # look for costing blocks
+            if hasattr(o, "costing") and hasattr(o.costing, "total_plant_cost"):
+                print(
+                    "%s: $%.2f Million" % (value(o.name), value(o.costing.total_cost))
+                )
+
+    def display_bare_erected_costs(fs):
+        print("-----Bare Erected Costs-----")
+        for o in fs.component_objects(descend_into=False):
+            # look for costing blocks
+            if hasattr(o, "costing") and hasattr(o.costing, "bare_erected_cost"):
+                print(
+                    "%s: $%.2f Million"
+                    % (value(o.name), value(o.costing.bare_erected_cost))
+                )
+
+    def display_equipment_costs(fs):
+        print("-----Equipment Costs-----")
+        for o in fs.component_objects(descend_into=False):
+            # look for costing blocks
+            if hasattr(o, "costing") and hasattr(o.costing, "equipment_cost"):
+                print(
+                    "%s: $%.2f Million"
+                    % (value(o.name), value(o.costing.equipment_cost))
+                )
+
+    def get_total_TPC(b, CE_index_year):
+        # This method accepts either a concrete model or a flowsheet block
+        # isinstance( m.fs.boiler.costing, QGESSCostingData)
+
+        try:
+            CE_index_units = getattr(
+                pyunits, "MUSD_" + CE_index_year
+            )  # millions of USD, for base year
+        except AttributeError:
+            raise Exception(
+                "CE_index_year %s is not a valid currency base option. "
+                "Valid CE index options include CE500, CE394 and years from "
+                "1990 to 2020." % (CE_index_year)
+            )
+
+        TPC_list = []
+        for o in b.parent_block().component_objects(descend_into=True):
+            # look for costing blocks
+            if hasattr(o, "costing") and hasattr(o.costing, "total_plant_cost"):
```

Comment:
If the user doesn't name a `UnitModelCostingBlock` m.fs.something.costing we won't be able to find it with this method.  Same with the other methods here that search for blocks named costing.

Reply:
The FlowsheetCostingBlock registers all costing blocks attached to it, so you should search for things via the list of registered units, not by name.

Reply:
Thanks for the suggestion, I will update places manually looking for "costing" attributes to look in the `_registered_unit_costing` list instead.